### ISSUE TITLE
ClassCastException ParserRuleRefNode cannot be cast to LexerRuleRefNode

### ISSUE THREAD

**Issue by Tomasu** [antlr/intellij-plugin-v4#351]
I have a hunch this is because I'm using split grammer files. eg: separate parser and lexer files.

```
java.lang.ClassCastException: org.antlr.intellij.plugin.psi.ParserRuleRefNode cannot be cast to org.antlr.intellij.plugin.psi.LexerRuleRefNode
	at org.antlr.intellij.plugin.psi.ModeSpecNode.getId(ModeSpecNode.java:36)
	at org.antlr.intellij.plugin.psi.MyPsiUtils$2.isAccepted(MyPsiUtils.java:84)
	at com.intellij.psi.search.PsiElementProcessor$CollectFilteredElements.execute(PsiElementProcessor.java:88)
	at com.intellij.psi.util.PsiTreeUtil$4.visitElement(PsiTreeUtil.java:876)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:70)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:27)
	at com.intellij.util.WalkingState.walkChildren(WalkingState.java:65)
	at com.intellij.util.WalkingState.elementStarted(WalkingState.java:52)
	at com.intellij.psi.PsiWalkingState.elementStarted(PsiWalkingState.java:79)
	at com.intellij.psi.PsiRecursiveElementWalkingVisitor.visitElement(PsiRecursiveElementWalkingVisitor.java:48)
	at com.intellij.psi.util.PsiTreeUtil$4.visitElement(PsiTreeUtil.java:877)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.intellij.psi.util.PsiTreeUtil.processElements(PsiTreeUtil.java:872)
	at com.intellij.psi.util.PsiTreeUtil.collectElements(PsiTreeUtil.java:848)
	at org.antlr.intellij.plugin.psi.MyPsiUtils.findSpecNode(MyPsiUtils.java:89)
	at org.antlr.intellij.plugin.psi.GrammarElementRef.resolve(GrammarElementRef.java:93)
	at com.intellij.codeInsight.intention.AddAnnotationPsiFix.getContainer(AddAnnotationPsiFix.java:73)
	at com.intellij.codeInspection.dataFlow.EditContractIntention.getTargetMethod(EditContractIntention.java:46)
	at com.intellij.codeInspection.dataFlow.EditContractIntention.isAvailable(EditContractIntention.java:56)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.availableFor(ShowIntentionActionsHandler.java:127)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.lambda$getActionsToShow$1(ShowIntentionsPass.java:284)
	at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.chooseBetweenHostAndInjected(ShowIntentionActionsHandler.java:152)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:282)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:254)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:220)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:69)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:423)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1168)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:416)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:415)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:391)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:147)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:222)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:389)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:161)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by bjansen** [comment]
I think you have a `mode` with a lowercase name somewhere in your grammar, and I didn't test that case when I implemented navigation. Anyway, this will be fixed in 1.10. Thanks for reporting the bug!

**Comment by Tomasu** [comment]
I think putting the grammar files into the source root directly fixed the issue. Not 100% sure, I can recheck if you like. That however broke intellij's ability to properly auto complete uses of the generated code. Kindof annoying.
